core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -311,7 +313,11 @@ private[chisel3] object Converter {
         case (false, SpecifiedDirection.Flip | SpecifiedDirection.Input) =>
           fir.Field(getRef(elt, info).name, fir.Flip, extractType(elt, false, info))
       }
-      fir.BundleType(d.elements.toIndexedSeq.reverse.map { case (_, e) => eltField(e) })
+      if (!d.opaqueType)
+        fir.BundleType(d.elements.toIndexedSeq.reverse.map { case (_, e) => eltField(e) })
+      else
+        extractType(d.elements.head._2, true, info)
```

Comment:
should d.elements have only one item? Should we put an assertion for that?

Reply:
That's a good idea, I'll also add a unit test that should throw when user overrides `opaqueType` incorrectly